### ISSUE TITLE
JSON reporter produces invalid JSON if errors are spread across several files

### ISSUE THREAD

**Issue by cexbrayat** [palantir/tslint#1272]
### Bug Report 

- __TSLint version__: `3.10.2`
- __TypeScript version__: `18.10`
- __Running TSLint via__:  CLI

#### TypeScript code being linted

user.ts
```ts
class User {
  sayHello() {
    let unused = 'hello';
  }
}
```

pony.ts
```ts
class Pony {
  sayHello() {
    console.log('hello')
  }
}
```

with no `tslint.json`.

#### Actual behavior

Running `tslint *.ts -t json` produces (line return is mine to emphasize the issue): 

```
[{"endPosition":{"character":23,"line":2,"position":51},"failure":"' should be \"","name":"pony.ts","ruleName":"quotemark","startPosition":{"character":16,"line":2,"position":44}},{"endPosition":{"character":24,"line":2,"position":52},"failure":"missing semicolon","name":"pony.ts","ruleName":"semicolon","startPosition":{"character":24,"line":2,"position":52}}]
[{"endPosition":{"character":24,"line":2,"position":52},"failure":"' should be \"","name":"user.ts","ruleName":"quotemark","startPosition":{"character":17,"line":2,"position":45}}]
```

As you can see the JSON produced is invalid (2 arrays side by side, without being wrapped in an array and be separated by a comma).

#### Expected behavior

Each file produces an array of errors, instead of having one array for all errors (I guess...), which would be valid JSON.

**Comment by JKillian** [comment]
I know 😢 , see https://github.com/palantir/tslint/issues/656
